pilot/pkg/serviceregistry/kube/controller/ambientindex.go

```diff
@@ -823,17 +947,24 @@ func (c *Controller) getPodsInService(svc *v1.Service) []*v1.Pod {
 	return c.podsClient.List(svc.Namespace, klabels.ValidatedSetSelector(svc.Spec.Selector))
 }
 
-// PodInformation returns all WorkloadInfo's in the cluster.
+// AddressInformation returns all AddressInfo's in the cluster.
 // This may be scoped to specific subsets by specifying a non-empty addresses field
-func (c *Controller) PodInformation(addresses sets.Set[types.NamespacedName]) ([]*model.WorkloadInfo, []string) {
+func (c *Controller) AddressInformation(addresses sets.Set[types.NamespacedName]) ([]*model.AddressInfo, []string) {
 	if len(addresses) == 0 {
 		// Full update
 		return c.ambientIndex.All(), nil
 	}
-	var wls []*model.WorkloadInfo
+	var wls []*model.AddressInfo
 	var removed []string
 	for p := range addresses {
-		wl := c.ambientIndex.Lookup(p.Name)
+		wname := p.Name
+		// GenerateDeltas has the formatted wname from the xds request, but not sure if other callers
+		// have the format enforced
+		if found := strings.Count(p.Name, "/"); found == 0 {
```

Comment:
nit; should we use `strings.Cut()` here like before to be consistent?

Reply:
done